.travis.yml

```diff
@@ -21,35 +36,26 @@ jobs:
     #    - choco install python --version 3.8.0
     #    - python -m pip install --upgrade pip
     #  env: PATH=/c/Python38:/c/Python38/Scripts:$PATH
+    - name: "Deploy to pypi"
+      stage: deploy
+      script:
+        - >
+          if [[ -n "$TRAVIS_TAG" ]] && [[ "${TRAVIS_TAG}" != "$(<VERSION)" ]]; then
+              >&2 echo "ERROR: You have marked a release by applying a Git tag and your VERSION file doesn't match it"
+              >&2 echo "Tip: Don't tag if you want to deploy only to test.pypi.org"
+              false
+          fi
+        - export VERSION=${TRAVIS_TAG:-$(<VERSION).dev${TRAVIS_BUILD_NUMBER}}
```

Comment:
it isn't necessarily a problem, but it might be - just worth a check: will VERSION populate correctly to the deployer if it's not a `before_deploy`

https://docs.travis-ci.com/user/job-lifecycle/ unfortunately doesn't answer that